src/Open3D/Visualization/Rendering/Filament/FilamentScene.cpp

```diff
@@ -416,6 +418,20 @@ void FilamentScene::SetLightColor(const LightHandle& id,
     }
 }
 
+Eigen::Vector3f FilamentScene::GetLightDirection(const LightHandle& id) const {
+    const auto found = entities_.find(id);
+    if (found != entities_.end()) {
+        auto& lightManager = engine_.getLightManager();
+        filament::LightManager::Instance inst =
+                lightManager.getInstance(found->second.info.self);
+        if (lightManager.isDirectional(inst)) {
```

Comment:
I'm not sure, but it would fails for Spot lights. Why should we even check light type here? It would just return incorrect value, if light is of inappropriate type.

Reply:
Well, I tend to like to know what the incorrect value that I am returning is, but I guess it's an error either way, so if you're okay with just returning what Filament gives us, I guess I am.